.github/workflows/ci.yml

```diff
@@ -130,16 +142,31 @@ jobs:
             cxxstd: "03,11,14,17,2a"
             os: macos-10.15
 
-    runs-on: ${{matrix.os}}
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
 
     steps:
+      - name: Check if running in container
+        if: matrix.container != ''
+        run: echo "GHA_CONTAINER=${{ matrix.container }}" >> $GITHUB_ENV
+      - name: If running in container, upgrade packages
+        if: matrix.container != ''
+        run: |
+            apt-get -o Acquire::Retries=3 update && DEBIAN_FRONTEND=noninteractive apt-get -y install tzdata && apt-get -o Acquire::Retries=3 install -y sudo software-properties-common wget curl apt-transport-https make apt-file sudo unzip libssl-dev build-essential autotools-dev autoconf automake g++ libc++-helpers python ruby cpio gcc-multilib g++-multilib pkgconf python3 ccache libpython-dev
+            sudo apt-add-repository ppa:git-core/ppa
```

Comment:
Why install git from PPA?

Reply:
Why install git from PPA? This CI script was designed to run on all the target operating systems including ubuntu LTS 12.04, 14.04, 16.04 etc. Those versions have quite out-of-date versions of git. This strategy, of installing newer utility packages such as git on older OS versions seems to be common with CI providers. Both Github Actions and Travis are doing that: they are providing the user with _newer_ versions of compilers, shells, etc.